### PR TITLE
feat: update button component

### LINKED ISSUE #18: Button: Create button component
[Figma Design](https://www.figma.com/file/msxmKajRPLYiDDmsP60Ndh/Design-System-v1.1?type=design&node-id=13-37022&t=ttVz3vzOxnYJ4iYn-4)

should support:

- [x] type: text - for buttons without background or shadow
- [x] different colors and states
- [x] outline button
- [x] rounded, or tiled button
- [x] light and dark mode colors
- [x] support icon only
- [x] add loading state (need to make the spinner component first)

### PR DESCRIPTION
**Pull request by lukicenturi**
Part of #18

### CODE REVIEW

File: src/components/buttons/Button.vue [hunk 1]
```diff
@@ -1,49 +1,74 @@
...
         [css.loading]: loading,
-        [css.tile]: tile,
+        [css._rounded]: rounded,
+        [css.fab]: fab,
+        [css.icon]: icon,
         [css.text]: text,
-        [css.sm]: sm,
-        [css.lg]: lg,
+        [css.sm]: size === 'sm',
+        [css.lg]: size === 'lg',
```
Comment (tewshi) [inline src/components/buttons/Button.vue:71]: similar to `color`, I think we can do
`css[size ?? ''],`

**Review by tewshi — APPROVED** [review]
lgtm!

File: src/components/buttons/Button.vue [hunk 2]
```diff
@@ -1,49 +1,73 @@
...
-withDefaults(
+const props = withDefaults(
   defineProps<{
     disabled?: boolean;
     loading?: boolean;
     outlined?: boolean;
     color?: ContextColorsType;
-    tile?: boolean;
-    elevated?: boolean;
+    rounded?: boolean;
```
Comment (kelsos) [inline src/components/buttons/Button.vue:10]: do you think we can introduce variants here too?

Comment (lukicenturi) [reply src/components/buttons/Button.vue:10]: yeah I think so. 
The variant can be either `default` | `outlined` | `text`
the other prop can be used simultaneously

Comment (kelsos) [reply src/components/buttons/Button.vue:9]: Can fab be mixed with other variants?

Comment (lukicenturi) [reply src/components/buttons/Button.vue:9]: fab can be used with icon, but not with outlined | text I think. Okay I can put it in variant too.

**Pull request merged**